v2.0/getting-started/docker/tutorials/advanced-policy.md

```diff
@@ -2,229 +2,292 @@
->   labels:
->    foo: bar
-> spec:
->   tags:
->   - tag1
->   - tag2s
->   ingress:
->   - action: deny
->     protocol: tcp
->     icmp:
->        type: 10
->        code: 6
->     notProtocol: udp
->     notICMP:
->        type: 19
->        code: 255
->     source:
->       tag: production
->       net: 10.0.0.0/16
->       selector: type=='application'
->       ports:
->       - 1234
->       - "10:20"
->       notTag: bartag
->       notNet: 10.1.0.0/16
->       notSelector: type=='database'
->       notPorts:
->       - 1050
->     destination:
->       tag: alphatag
->       net: 10.2.0.0/16
->       selector: type=='application'
->       ports:
->       - "100:200"
->       notTag: bananas
->       notNet: 10.3.0.0/16
->       notSelector: type=='apples'
->       notPorts:
->       - "105:110"
->   egress:
->   - action: allow
->     source:
->       selector: type=='application'
-> EOF
-```
+#### 2. Create the profiles
 
-#### View the policy associated with the network
+Create the profiles for each of these networks.  The profile may contains a set 
+of labels and a set of ingress and egress rules and actions where each rule can 
+filter packets based on a variety of source or destination attributes (including 
+selector based filtering using label selection).  The labels and rules are applied
+directly to each container in the corresponding network.  
 
-You can use the `calicoctl get -o yaml profile <profile>` to display the
-rules in the profile associated with the `databases` network.
+Use `calicoctl apply` to create or update the profiles:
 
 ```
-$ calicoctl get profile databases -o yaml
+cat << EOF | dist/calicoctl apply -f -
 - apiVersion: v1
   kind: profile
   metadata:
-    labels:
-      foo: bar
     name: databases
+    labels:
+      role: database
   spec:
-    egress:
+    ingress:
     - action: allow
-      destination: {}
+      protocol: tcp
       source:
-        selector: type=='application'
-    ingress:
-    - action: deny
+        selector: role == 'app'
       destination:
-        net: 10.2.0.0/16
-        notNet: 10.3.0.0/16
-        notPorts:
-        - 105:110
-        notSelector: type=='apples'
-        notTag: bananas
         ports:
-        - 100:200
-        selector: type=='application'
-        tag: alphatag
-      icmp:
-        code: 6
-        type: 10
-      notICMP:
-        code: 255
-        type: 19
-      notProtocol: udp
-      protocol: tcp
+        -  3306
+    - action: allow
       source:
-        net: 10.0.0.0/16
-        notNet: 10.1.0.0/16
-        notPorts:
-        - 1050
-        notSelector: type=='database'
-        notTag: bartag
-        ports:
-        - 1234
-        - "10:20"
-        selector: type=='application'
-        tag: production
-    tags:
-    - tag1
-    - tag2s
+        selector: role == 'database'
+    egress:
+    - action: allow
+      destination:
+        selector: role == 'database'
+- apiVersion: v1
+  kind: profile
+  metadata:
+    name: apps
```

Comment:
yes yes yes!   good idea ... i don't know why i always choose app since i always end up changing it!